LibertyDSNP/sdk-ts
# Pre 0.1.0 Cleanup, Fixing Exports

Problem
=======
Multimodule require and imports were not working

Solution
========

Cleanup and fix multimodule support along with various other cleanups

with @rosalinekarr, @aneyzberg 

Change summary:
---------------
* Support recursive multimodules
* Disable baseUrl
* Fix readme examples
* Update `@dsnp` packages to latest version instead of `@next` versions
* Update typedoc settings to hide most of the internals

Steps to Verify:
----------------
For Multimodules all the following should now work:

```typescript
import * as rootStar from "@dsnp/sdk";
console.log({ rootStar });

import * as coreStar from "@dsnp/sdk/core";
console.log({ coreStar });

import * as apub from "@dsnp/sdk/core/activityPub";
console.log({ coreActivityPubStar: apub });

import { activityPub } from "@dsnp/sdk/core";
console.log({ coreDeconstructed: activityPub });

import { getConfig } from "@dsnp/sdk";
console.log({ rootDeconstructed: getConfig });

import { validateReply } from "@dsnp/sdk/core/activityPub";
console.log({ coreActivityPubDeconstructed: validateReply });
```

```js
const rootStar = require("@dsnp/sdk");
console.log({ rootStar });

const coreStar = require("@dsnp/sdk/core");
console.log({ coreStar });

const apub = require("@dsnp/sdk/core/activityPub");
console.log({ coreActivityPubStar: apub });

const { activityPub } = require("@dsnp/sdk/core");
console.log({ coreDeconstructed: activityPub });

const { getConfig } = require("@dsnp/sdk");
console.log({ rootDeconstructed: getConfig });

const { validateReply } = require("@dsnp/sdk/core/activityPub");
console.log({ coreActivityPubDeconstructed: validateReply });
```
